pyproject.toml

```diff
@@ -0,0 +1,102 @@
+[build-system]
+requires = ["flit-gettext", "flit_scm", "wheel"]
+build-backend = "flit_gettext.scm"
+
+[project]
+name = "django-hijack"
+authors = [
+  { name = "arteria GmbH", email = "admin@arteria.ch" },
+  { name = "Johannes Maron", email = "johannes@maron.family" },
+]
+readme = "README.md"
+license = { file = "LICENSE" }
+keywords = ["Markdown", "Django", "email", "templates", "HTML"]
+dynamic = ["version", "description"]
+classifiers = [
+  "Development Status :: 5 - Production/Stable",
+  "Environment :: Web Environment",
+  "License :: OSI Approved :: MIT License",
+  "Intended Audience :: Developers",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Framework :: Django",
+  "Framework :: Django :: 4.2",
+  "Framework :: Django :: 5.0",
+]
+requires-python = ">=3.8"
+dependencies = ["django>=4.2"]
+
+[project.optional-dependencies]
+test = [
+  "pytest",
+  "pytest-cov",
+  "pytest-django",
+]
+lint = [
+  "black==24.4.2",
```

Comment:
Actually, ruff has formatter which is black-compatible. 
https://docs.astral.sh/ruff/formatter/
For small projects, it is fairly good.

Reply:
Have you done this before? I don't understand the docs, how do I configure the black rule set? Or is it just a formatter? How do I check for the black ruleset?

Reply:
Yes. I use it in my main project as a replacement for black. 
From doc:
> The Ruff formatter is an extremely fast Python code formatter **designed as a drop-in replacement** for [Black](https://pypi.org/project/black/), available as part of the ruff CLI via ruff format.

There might be small changes but in general, it is the same.

```
[tool.ruff]
line-length = 88
indent-width = 4
target-version = "py311" # Fix this

[tool.ruff.format]
# Like Black, use double quotes for strings.
quote-style = "double"
# Like Black, indent with spaces, rather than tabs.
indent-style = "space"
# Like Black, respect magic trailing commas.
skip-magic-trailing-comma = false
# Like Black, automatically detect the appropriate line ending.
line-ending = "auto"
preview = true # if you want to enable preview mode
```

In general I can see astral utilities becoming defacto standards. As an example, [one of the best Django project templates](https://github.com/cookiecutter/cookiecutter-django) (in my opinion) has already fully switched to ruff.

Reply:
Also, I use actions in GH

```
  ruff:
    runs-on: ubuntu-20.04
    timeout-minutes: 3
    steps:
      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29
      - name: Get ruff version from requirements.in
        run: echo "RUFF_VERSION=$(grep -IEio '^ruff==(\S+)' requirements.in | awk -F '==' '{print $2}')" >> $GITHUB_ENV
      - uses: chartboost/ruff-action@v1
        with:
          version: ${{ env.RUFF_VERSION }}

  ruff-format:
    runs-on: ubuntu-20.04
    timeout-minutes: 3
    steps:
      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29
      - name: Get ruff version from requirements.in
        run: echo "RUFF_VERSION=$(grep -IEio '^ruff==(\S+)' requirements.in | awk -F '==' '{print $2}')" >> $GITHUB_ENV
      - uses: chartboost/ruff-action@v1
        with:
          version: ${{ env.RUFF_VERSION }}
          args: 'format --check'
```

Reply:
I'd rather not introduce another 3rd party dependency into the stack, considering this package deals with session security. Fewer dependencies are probably better. But thanks, I'll check how we can use all of this in other projects. Appreciate the samples. Also love the `py311  # fix this` comment :P